src/pfe/file-watcher/server/test/scripts/exec.sh

```diff
@@ -37,31 +38,37 @@ function setup {
     TEST_OUTPUT=$TEST_OUTPUT_DIR/test_output.xml
     TEST_LOG=$TEST_OUTPUT_DIR/$TEST_SUITE-test.log
     CODEWIND_CONTAINER_ID=$(docker ps | grep codewind-pfe-amd64 | cut -d " " -f 1)
+    PROJECTS_CLONE_DATA_FILE="./resources/projects-clone-data"
 
     mkdir -p $TEST_OUTPUT_DIR \
     && docker exec -it $CODEWIND_CONTAINER_ID bash -c "cd /file-watcher/server; npm install --only=dev" \
     && docker cp $TEST_DIR/test $CODEWIND_CONTAINER_ID:$TEST_DIR_CONTAINER
 
-    if [[ $? -ne 0 ]]; then
+    if [[ ($? -ne 0) ]]; then
         echo -e "${RED}Test setup failed. ${RESET}\n"
         exit 1
     fi
-    
+
+    # Clean up workspace if needed
+    if [[ ($CLEAN_WORKSPACE == "y") ]]; then
+        echo -e "${BLUE}Cleaning up workspace. ${RESET}\n"
+        rm -rf $CW_DIR/codewind-workspace/*
+    fi
+
     # Clone projects to workspace
-    declare -a PROJECT_URLS
-    PROJECT_URLS["codewindtestpython"]="https://github.com/microclimate-dev2ops/SVTPythonTemplate"
-    PROJECT_URLS["codewindtestgo"]="https://github.com/microclimate-dev2ops/microclimateGoTemplate"
-    PROJECT_URLS["codewindtestlagom"]="https://github.com/microclimate-dev2ops/lagomJavaTemplate"
-    PROJECT_URLS["codewindtestspring"]="https://github.com/microclimate-dev2ops/springJavaTemplate"
-    PROJECT_URLS["codewindtestmicroprofile"]="https://github.com/microclimate-dev2ops/javaMicroProfileTemplate"
-    PROJECT_URLS["codewindtestnodejs"]="https://github.com/microclimate-dev2ops/nodeExpressTemplate.git"
-    PROJECT_URLS["codewindtestswift"]="https://github.com/microclimate-dev2ops/swiftTemplate"
-    
-    echo -e "${BLUE}Cloning projects to $CW_DIR/codewind-workspace ${RESET}\n"
-    for PROJECT_NAME in ${!PROJECT_URLS[@]}; do
-        echo -e "${BLUE}Cloning ${PROJECT_URLS[$PROJECT_NAME]}. ${RESET}\n"
-        clone $PROJECT_NAME $CW_DIR/codewind-workspace ${PROJECT_URLS[$PROJECT_NAME]}
-    done
+    echo -e "${BLUE}Cloning projects to $CW_DIR/codewind-workspace. ${RESET}"
+    while IFS= read -r LINE; do
+        PROJECT_NAME=$(echo $LINE | cut -d "=" -f 1)
+        PROJECT_URL=$(echo $LINE | cut -d "=" -f 2)
+        echo -e "\n\nProject name is: $PROJECT_NAME, project URL is $PROJECT_URL"
+        echo -e "${BLUE}Cloning $PROJECT_URL. ${RESET}"
+        clone $PROJECT_NAME $CW_DIR/codewind-workspace $PROJECT_URL
+
+        if [[ ($? -ne 0) ]]; then
+            echo -e "${RED}Test setup failed. ${RESET}\n"
```

Comment:
Can we update the message to be more specific? Eg. Cloning project $PROJECT_NAME failed.